testing/src/main/java/org/oppia/android/testing/FakePerformanceMetricUtils.kt

```diff
@@ -0,0 +1,102 @@
+package org.oppia.android.testing
+
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.util.logging.performancemetrics.PerformanceMetricsUtils
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** A test specific fake for performance metric utils. */
+@Singleton
+class FakePerformanceMetricUtils @Inject constructor() : PerformanceMetricsUtils {
+  /**
+   * Returns the apk size of the current device.
+   *
+   * It returns a default value unless a specific value has been set using [setApkSize].
+   */
+  internal var testApkSize = 0L
```

Comment:
For this & the others: given there are setters and getters, why can't these be private?

Also, are there any concerns with cross-thread access to this utility? Does it need to use atomics for these values?

Reply:
made these private and no, there aren't any cross-thread access concerns for this and I don't think there's any need for atomics here.